### PR TITLE
chore: update examples to iroh v0.29.0

### PR DESCRIPTION
**Pull request by ramfox**

### CODE REVIEW

File: extism/iroh-extism-host-functions/src/lib.rs [hunk 1]
```diff
@@ -22,36 +24,82 @@ pub async fn default_iroh_extism_data_root() -> Result<PathBuf> {
     Ok(path.join(IROH_EXTISM_DATA_DIR))
 }
 
-pub async fn create_iroh(path: PathBuf) -> Result<IrohNode> {
-    let node = iroh::node::Node::persistent(path).await?.spawn().await?;
-    Ok(node)
+pub struct Iroh {
```
Comment (matheus23) [inline extism/iroh-extism-host-functions/src/lib.rs:27]: We should probably avoid calling this `Iroh`, but that's just a minor nit.

Comment (matheus23) [reply extism/iroh-extism-host-functions/src/lib.rs:27]: How about `ExtismIroh`? Maybe `IrohBlobs`?

File: tauri-todos/src-tauri/src/iroh.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+pub(crate) type DocsClient = iroh_docs::rpc::client::docs::Client<
+    FlumeConnector<iroh_docs::rpc::proto::Response, iroh_docs::rpc::proto::Request>,
+>;
+
+pub(crate) struct Iroh {
+    _local_pool: Arc<LocalPool>,
+    router: Router,
+    pub(crate) blobs: BlobsClient,
+    pub(crate) docs: DocsClient,
+}
```
Comment (matheus23) [inline tauri-todos/src-tauri/src/iroh.rs:22]: Hm... Same thing here. Maybe we can just go with `Node`? :joy:

**Review by matheus23 — APPROVED** [review]
Thank you!!!

Just minor nits. I'd say they're optional. LMK if I should lend a hand :v:

File: extism/iroh-extism-host-functions/src/lib.rs (hunk 1 above)
Comment (dignifiedquire) [reply extism/iroh-extism-host-functions/src/lib.rs:27]: it's extism iroh struct, so seems fine to me, better than `Node` 😅

File: tauri-todos/src-tauri/src/iroh.rs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+        // add iroh gossip
+        let addr = builder.endpoint().node_addr().await?;
+        let gossip = iroh_gossip::net::Gossip::from_endpoint(
+            builder.endpoint().clone(),
+            Default::default(),
+            &addr.info,
+        );
+        builder = builder.accept(
+            iroh_gossip::net::GOSSIP_ALPN.to_vec(),
```
Comment (dignifiedquire) [inline tauri-todos/src-tauri/src/iroh.rs:59]: no need for `to_vec` and there is `iroh_gossip::ALPN` reexport now

File: tauri-todos/src-tauri/src/iroh.rs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+            Arc::new(gossip.clone()),
+        );
+
+        // add iroh blobs
+        let downloader = Downloader::new(
+            blobs_store.clone(),
+            builder.endpoint().clone(),
+            local_pool.handle().clone(),
+        );
+        let blobs = Arc::new(Blobs::new(
```
Comment (dignifiedquire) [inline tauri-todos/src-tauri/src/iroh.rs:69]: `Blobs::persistent` is nicer to use

File: tauri-todos/src-tauri/src/iroh.rs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+            blobs_store.clone(),
+            local_pool.handle().clone(),
+            EventSender::default(),
+            downloader.clone(),
+            builder.endpoint().clone(),
+        ));
+        builder = builder.accept(iroh_blobs::protocol::ALPN.to_vec(), blobs.clone());
```
Comment (dignifiedquire) [inline tauri-todos/src-tauri/src/iroh.rs:76]: no need for `to_vec` and there is `iroh_blobs::ALPN` reexport now

File: tauri-todos/src-tauri/src/main.rs [hunk 6]
```diff
@@ -26,30 +27,26 @@ async fn setup<R: tauri::Runtime>(handle: tauri::AppHandle<R>) -> Result<()> {
...
+    let iroh = Iroh::new(data_root).await?;
+    handle.manage(AppState::new(iroh));
 
     Ok(())
 }
 
 struct AppState {
     todos: Mutex<Option<(Todos, tokio::task::JoinHandle<()>)>>,
-    iroh: IrohNode,
+    iroh: Arc<Iroh>,
```
Comment (dignifiedquire) [inline tauri-todos/src-tauri/src/main.rs:38]: no need for the `Arc`, you can just derive `Clone` for `Iroh` if you need to

**Review by dignifiedquire — APPROVED** [review]
some nits, for simpler code, otherwise lgtm

File: extism/iroh-extism-host-functions/src/lib.rs (hunk 1 above)
Comment (ramfox) [reply extism/iroh-extism-host-functions/src/lib.rs:27]: I'm going to keep it `Iroh`, to me it's "all the iroh stuff".

**Pull request merged**